bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java

```diff
@@ -0,0 +1,314 @@
+		createButton(parent, IDialogConstants.CANCEL_ID, IDialogConstants.CANCEL_LABEL, false);
+	}
+
+	@Override
+	public boolean close()
+	{
+		if (espToolCommands.checkActiveFlashEraseProcess())
+		{
+			MessageBox messageBox = new MessageBox(getParentShell(), SWT.ICON_QUESTION | SWT.YES | SWT.NO | SWT.CANCEL);
+			messageBox.setMessage(Messages.EraseFlashDialog_EraseFlashInProcessMessageQuestion);
+			messageBox.setText(Messages.EraseFlashDialog_EraseFlashInProcessMessageTitle);
+			switch (messageBox.open())
+			{
+			case SWT.YES:
+				espToolCommands.killEraseFlashProcess();
+				break;
+			case SWT.CANCEL:
+				return false;
+			default:
+				break;
+			}
+		}
+		return super.close();
+	}
+
+	@Override
+	protected void okPressed()
+	{
+		setReturnCode(OK);
+		getButton(IDialogConstants.OK_ID).setEnabled(false);
+		deviceInformationText.setText("Erasing Flash....");
+		String selectedPort = comPortsCombo.getText();
+		Thread eraseFlashThread = new Thread(new Runnable()
+		{
+			@Override
+			public void run()
+			{
+				try
+				{
+					Display.getDefault().syncExec(() -> comPortsCombo.setEnabled(false));
+					Display.getDefault().syncExec(() -> deviceInformationText.setText(""));
+					Process eraseFlashProcess = espToolCommands.eraseFlash(selectedPort);
+					InputStream targetIn = eraseFlashProcess.getInputStream();
+					BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(targetIn));
+					String line = "";
+					while ((line = bufferedReader.readLine()) != null)
+					{
+						final String toWrite = line;
+						Display.getDefault().asyncExec(() -> deviceInformationText.append(toWrite + "\n"));
+					}
+				}
+				catch (Exception e)
+				{
+					UIPlugin.getDefault().getLog().error("Exception while getting erasing flash", e);
+				}
+
+				Display.getDefault().syncExec(() -> comPortsCombo.setEnabled(true));
+			}
+		});
+		if (espToolCommands.checkActiveFlashEraseProcess())
+		{
+			MessageBox messageBox = new MessageBox(getParentShell(), SWT.ICON_INFORMATION);
+			messageBox.setMessage(Messages.EraseFlashDialog_EraseFlashInProcessMessage);
+			messageBox.setText(Messages.EraseFlashDialog_EraseFlashInProcessMessageTitle);
+			messageBox.open();
+		}
+		else
+		{
+			eraseFlashThread.start();
+		}
+	}
+
+	@Override
+	protected Point getInitialSize()
+	{
+		return new Point(450, 300);
+	}
+
+	private class ComPortSelectionListener extends SelectionAdapter implements Runnable
+	{
+		private String selectedPort;
+		private Thread infoThread;
+
+		@Override
+		public void widgetSelected(SelectionEvent event)
+		{
+			getButton(IDialogConstants.OK_ID).setEnabled(true);
+			selectedPort = ((Combo) event.getSource()).getText();
+			try
+			{
+				deviceInformationText.setText(Messages.EraseFlashDialog_LoadingMessage);
+				infoThread = new Thread(this);
+				infoThread.start();
+			}
+			catch (Exception exception)
+			{
+				UIPlugin.getDefault().getLog().error("Exception while getting chip information", exception);
+			}
```

Comment:
externalize this error message. We can use our Logger class.

Reply:
Done